paddle/fluid/operators/mkldnn/activation_mkldnn_op.cc

```diff
@@ -169,47 +153,43 @@ void eltwise_grad(const framework::ExecutionContext &ctx,
 
   std::vector<int> diff_dst_tz = framework::vectorize2int(diff_y->dims());
 
+  // diff_dst and src dims should be the same
+  auto src_format =
+      diff_dst_tz.size() == 2 ? mkldnn::memory::format::nc : x->format();
+
   auto diff_y_format =
       diff_dst_tz.size() == 2 ? mkldnn::memory::format::nc : diff_y->format();
 
   auto diff_dst_md = platform::MKLDNNMemDesc(
       diff_dst_tz, platform::MKLDNNGetDataType<T>(), diff_y_format);
 
   std::string key = platform::MKLDNNHandler::GetHash(
-      diff_dst_tz, std::to_string(algorithm) + std::to_string(alpha) +
-                       std::to_string(beta) + ctx.op().Input("X"));
+      diff_dst_tz, std::to_string(algorithm) + std::to_string(src_format) +
```

Comment:
What do you think about moving the code responsible for creating the key for one function ?

Reply:
That's a good suggestions. I have updated changes